### PR TITLE
white screen when opening research plan

### PR DESCRIPTION
**Pull request by mekkyz**
Fixes https://github.com/ComPlat/chemotion_ELN/issues/1382

- null checks in to prevent errors from deleted samples/reactions while opening research plans

### CODE REVIEW

**Review by ptrxyz — DISMISSED** [review]

File: app/packs/src/apps/mydb/elements/list/ElementsTableEntries.js [hunk 1]
```diff
@@ -180,7 +161,7 @@ export default class ElementsTableEntries extends Component {
       case 13: // Enter
       case 39: // Right
         if (keyboardElementIndex != null && elements[keyboardElementIndex] != null) {
```
[the remark this answers is not in the document]
Comment (mekkyz) [reply app/packs/src/apps/mydb/elements/list/ElementsTableEntries.js:163]: Yes it makes no sense, but it was done before, not by me

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldSample.js [hunk 2]
```diff
@@ -51,7 +55,9 @@ class ResearchPlanDetailsFieldSample extends Component {
   componentDidUpdate() {
     const { field } = this.props;
     const { idle, sample } = this.state;
-    if (idle && field.value.sample_id !== sample.id && hasAuth(sample.id)) {
+
+    // check if field.value and sample are not undefined before accessing their properties
+    if (idle && field && field.value && sample && field.value.sample_id !== sample.id && hasAuth(sample.id)) {
```
[the remark this answers is not in the document]
Comment (mekkyz) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldSample.js:60]: Applied

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldSample.js [hunk 3]
```diff
@@ -114,18 +124,23 @@ class ResearchPlanDetailsFieldSample extends Component {
...
     const { sample } = this.state;
-    if (!hasAuth(sample.id)) {
+    if (sample && !hasAuth(sample.id)) {
       return noAuth(sample);
     }
     let className = 'drop-target';
     if (isOver) className += ' is-over';
     if (canDrop) className += ' can-drop';
-    return connectDropTarget(<div className={className}>{sample.id ? this.renderSample(sample) : 'Drop sample here.'}</div>);
+    return connectDropTarget(<div className={className}>{sample && sample.id ? this.renderSample(sample) : 'Drop sample here.'}</div>);
```
[the remark this answers is not in the document]
Comment (mekkyz) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldSample.js:133]: applied

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldSample.js [hunk 4]
```diff
@@ -80,7 +86,9 @@ class ResearchPlanDetailsFieldSample extends Component {
     if (edit) {
       link = (
         <p>
-          Sample: {title}
+          Sample:
+          {' '}
```
[the remark this answers is not in the document]
Comment (mekkyz) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldSample.js:90]: Applied

**Review by PiTrem — APPROVED** [review]

**Comment by PiTrem** [comment]
not working

**Pull request closed without merging**